src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -762,8 +877,51 @@ internal override void Complete()
                     }
 
                     this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
-                    if (_nativeProcess.ExitCode != 0)
-                        this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+                    if (_nativeProcess.ExitCode == 0)
+                    {
+                        return;
+                    }
+
+                    this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+
+                    bool nativeCommandUseErrorActionPreference = ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName)
+                        && this.Command.Context.GetBooleanPreference(
+                            SpecialVariables.PSNativeCommandUseErrorActionPreferenceVarPath,
+                            InitialSessionState.DefaultPSNativeCommandUseErrorActionPreference,
+                            out _);
+
+                    if (!nativeCommandUseErrorActionPreference)
```

Comment:
I'd prefer roll out the variable here and remove the help variable. This will simplify switching from experimental (less code changes).